aptos-move/move-examples/mint_nft_v2/2-Using-Resource-Account/sources/create_nft_with_resource_account.move

```diff
@@ -0,0 +1,201 @@
+        property_values: vector<vector<u8>>,
+        collection_object_address: address,
+    }
+
+    /// `init_module` is automatically called when publishing the module.
+    /// In this function, we create an example NFT collection and an example token.
+    fun init_module(resource_signer: &signer) {
+        let collection_name = string::utf8(b"Collection name");
+        let description = string::utf8(b"Description");
+        let collection_uri = string::utf8(b"Collection uri");
+        let token_name = string::utf8(b"Token name");
+        let token_uri = string::utf8(b"Token uri");
+        let maximum_supply = 1000;
+
+        aptos_token::create_collection(
+            resource_signer,
+            description,
+            maximum_supply,
+            collection_name,
+            collection_uri,
+            false, // mutable_description
+            false, // mutable_royalty
+            false, // mutable_uri
+            false, // mutable_token_description
+            false, // mutable_token_name
+            true, // mutable_token_properties
+            false, // mutable_token_uri
+            false, // tokens_burnable_by_creator
+            false, // tokens_freezable_by_creator
+            5, // royalty_numerator
+            100, // royalty_denominator
+        );
+
+        // Retrieve the resource signer's signer capability and store it within the `ModuleData`.
+        // Note that by calling `resource_account::retrieve_resource_account_cap` to retrieve the resource account's signer capability,
+        // we rotate the resource account's authentication key to 0 and give up our control over the resource account. Before calling this function,
+        // the resource account has the same authentication key as the source account so we had control over the resource account.
+        let resource_signer_cap = resource_account::retrieve_resource_account_cap(resource_signer, @source_addr);
+        let resource_address = signer::address_of(resource_signer);
+        let collection_object_address = collection::create_collection_address(&resource_address, &collection_name);
+
+        // Store the token data id and the resource account's signer capability within the module, so we can programmatically
+        // sign for transactions in the `mint_event_ticket()` function.
+        move_to(resource_signer, ModuleData {
+            signer_cap: resource_signer_cap,
+            collection_name,
+            token_description: string::utf8(b""),
+            token_name,
+            token_uri,
+            property_keys: vector<String>[string::utf8(b"given_to")],
+            property_types: vector<String>[ string::utf8(b"address") ],
+            property_values: vector<vector<u8>>[bcs::to_bytes(&@source_addr)],
+            collection_object_address,
+        });
+    }
+
+    /// Mint an NFT to the receiver. Note the difference from the tutorial in part 1, here we only ask for the receiver's
```

Comment:
We technically need the receiver's signer here, not just address. The real diff between this and step 1 code is that this function can now be called by the user/receiver instead of the admin

Reply:
if I'm not misunderstanding what you mean, I meant to just clarify that you can send the token object without the `receiver`s approval (signer) now, and the only reason we have the receiver as a signer is so they're approving of receiving the mint. I can explain it/word it better in the tutorial

Reply:
Explaining it better in the tutorial makes sense